infusionvlc/somnium_server
# Create Tag and TagEntity

### :pushpin: References
* **Issue:** #7 

### :tophat: What is the goal?

Tag a dream so they can be classified, also when a dream is updated the tags need to be updated too.

### :memo: How is it being implemented?

* Create `Tag` and `TagEnity` models.
* Create `FindTagByTitle` and `GetOrCreateTag` usecases.
* Create `GetOrCreateTagTest`.
* Add logic to add tags when a `Dream` is created.
* Add logic to update tags when a `Dream` is updated.
* Modify `CreateDreamTest` and `EditDreamTest` to work with the new `GetOrCreateTag` use case.

### :boom: How can it be tested?

* **Case 1**: Create a `Tag` when a `Dream` is created:
  * Call the endpoint `POST localhost:8080/dreams/v1/` and in the `tags` param specify all the tags that you want to create. (Example: `tags: ["tag1", "tag2]`)
  * You can create more than one `Dream` with different tags to check if the tags that already exist are not created again.
* **Case 2**: Create a `Tag` when a `Dream` is updated:
  * Call the endpoint `PUT localhost:8080/dreams/v1/{id}` and in the `tags` param specify all the tags that you want to create. (Example: `tags: ["tag1", "tag2]`)
 * You can create new tags to check that if the tags that already exist are not created again.
 
### :floppy_disk: Requires DB migration?
